src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqBurdenEstimateRepository.kt

```diff
@@ -272,28 +275,30 @@ class JooqBurdenEstimateRepository(
 
         if (set.status == BurdenEstimateSetStatus.COMPLETE)
         {
-            throw OperationNotAllowedError("This burden estimate set has been marked as complete." +
+            throw InvalidOperationError("This burden estimate set has been marked as complete." +
                     " You must create a new set if you want to upload any new estimates.")
         }
 
-        if (type == BurdenEstimateSetTypeCode.STOCHASTIC)
-        {
-            stochasticBurdenEstimateWriter.addEstimatesToSet(setId, estimates, responsibilityInfo.disease)
-        }
-        else
-        {
-            centralBurdenEstimateWriter.addEstimatesToSet(setId, estimates, responsibilityInfo.disease)
-        }
-
+        getEstimateWriter(set).addEstimatesToSet(setId, estimates, responsibilityInfo.disease)
         changeBurdenEstimateStatus(setId, BurdenEstimateSetStatus.PARTIAL)
         updateCurrentBurdenEstimateSet(responsibilityInfo.id, setId, type)
     }
 
     override fun closeBurdenEstimateSet(setId: Int, groupId: String, touchstoneId: String, scenarioId: String)
     {
         // Check all the IDs match up
-        getSetFromResponsibilityPath(groupId, touchstoneId, scenarioId, setId)
-        changeBurdenEstimateStatus(setId, BurdenEstimateSetStatus.COMPLETE)
+        val (set, _) = getSetFromResponsibilityPath(groupId, touchstoneId, scenarioId, setId)
+        val helper = getEstimateWriter(set)
+        val isEmpty = dsl.fetchAny(helper.table, helper.setField.eq(setId)) == null
```

Comment:
how about defining this as a function of `setId` on the estimate writer itself? I think fine to leave the naming as `...Writer` rather than `...Helper` though

Reply:
Yeah, I wasn't sure if adding that was a good idea, because of polluting the purpose of `BurdenEstimateWriter` away from writing. But I think you're right. Done here: https://github.com/vimc/montagu-api/pull/233/commits/652e1ac19c7881b383a23e520d502acffed50470